### DISCUSSION TITLE
17HS8401 Steppers Accuracy

### DISCUSSION THREAD

**Discussion by dwaterford75** [arkypita/LaserGRBL#1357] [Q&A]
Thanks in advance for anyone who contributes to my question.

I have just built a rig with a X, Y and Z axis all driven by 17HS8401 steppers.  My Y axis has two steppers on either side driving my gantry back and forth.

My power supply is a 24V 10A unit.  I have disconnected the Z Axis and connected my laser to my Arduino UNO.  Everything works ...... almost perfectly ... but not.

My first setup had a Arduino UNO with a CNC Shield and DR8825 stepper drivers.  I then changed this setup over to the same Arduino but TB6600 Stepper Motor Drivers.

When my machine is running, I find that it looses position on the X and Y axis.  So it will start working and if I stop it after a minute and ask it to return to origin, it will be off position.  So, if I draw a square, the start point and the end point will not meet up.  It is almost like the steppers are moving their steps too hard and I cannot get them to move slower and smoother.  I have tried various speeds but if I make it too slow, it seems the steppers miss steps and if I make it faster it is like the inertia of the system makes the motor miss as well.

My question is, what settings should I be using?
XYZ speed, acceleration, steps and any other settings within lasergrbl such as step pulse time, idle delay, etc.

Any assistance is greatly appreciated.

**Comment by StuartB4** [comment]
I use DRV8825's. Never used TB6600 but it sounds like you need to calibrate the steps/mm for the motors.
Or adjust the microstepping dip switches. 

You may already have this but if not it may help.  It's a pdf of the instruction manual for those drivers.
https://u.pcloud.link/publink/show?code=XZbgzfXZWhteetWK4B52BvLC5DW3tS2UX5mk

Or there is an online calibrator here: https://jscalc.io/calc/tHjfIiKMKjDF42Hg?fbclid=IwAR219p6S0qWzVAsY8aLC8nyQaXaMpfa-mSTNCIXA7TAMPa3VzrGoCDcimsI

You will need a ruler to measure how much the axis move when you jog them.
https://jscalc.io/calc/tHjfIiKMKjDF42Hg?fbclid=IwAR219p6S0qWzVAsY8aLC8nyQaXaMpfa-mSTNCIXA7TAMPa3VzrGoCDcimsI

Before you start goto grbl configuration on LaserGRBL Grbl menu and get the numbers for $100 and $101

This shows you how to use it. 
![Calibrator](https://user-images.githubusercontent.com/16980026/109965420-da328b80-7ce6-11eb-8b11-5e3f90472a26.jpg)

**Comment by dwaterford75** [comment]
Hi Stuart,
Fist off, thanks for the response.
I have played with many settings on the steppers.  I had the step count per mm far too high so it was really behaving badly.  I reduced this to around 20 and now the steppers move the accurate distance.  So if my image is 100x100, she moves darn close to 100mm.  By example, if I use the lasergrbl interface movement arrows on the bottom right, I get smooth accurate movement and the distance covered looks bang on to me.  This works on the X and the Y axis.  If I then push the play button, all hell breaks loose.  It is like lasergrbl is making it move in a lot of small steps so movement becomes quite rough and because its both X and Y and in both directions, I get this issue where the machine looses its home point.  I don't think it is missing steps or something, I also don't think the calibration is off.  It almost looks like the speed setting in lasergrbl is too high so it is moving the laser too fast.  Having said that, if I slow it down it also fails.

![20210310_101728](https://user-images.githubusercontent.com/63294939/110598199-2750aa80-818a-11eb-9468-fd67215668ae.jpg)

If you take a look at this photo, note the bottom middle of the logo.  You can see the starting point and the ending point are just not meeting up.

![20210310_101748](https://user-images.githubusercontent.com/63294939/110598372-5830df80-818a-11eb-9680-198ff3795f2a.jpg)

You can see the repeat of the problem in the text where the central part of the a is off.

![20210310_101807](https://user-images.githubusercontent.com/63294939/110598453-70a0fa00-818a-11eb-85e9-4b99929615cf.jpg)

Or on this one where you can clearly see the drift on the letters at the bottom.

Bear in mind, my rig looks like this so I am using dual stepper Y axis, single stepper X axis.  Both with Lead Screws.

![20210310_102406](https://user-images.githubusercontent.com/63294939/110598912-fde44e80-818a-11eb-808e-427c2494d090.jpg)
![20210310_102414](https://user-images.githubusercontent.com/63294939/110598923-0046a880-818b-11eb-97d6-7cd8a71310a9.jpg)
![20210310_102431](https://user-images.githubusercontent.com/63294939/110598930-02a90280-818b-11eb-8688-9a055e4c28c3.jpg)

I thought my design would be pretty infallible in terms of rigidity and precision but ...

I have disassemble things a little to pack it away.

Any assistance is greatly appreciated.

**Comment by StuartB4** [comment]
Ok. 
Can you start LaserGRBL and click Connect and go to Grbl configuration on the Grbl menu and get a screenshot or photo of the settings list and post it here. Thanks.

**Comment by StuartB4** [comment]
Don't know about TB6600 drivers, never used them. I use an Auduino UNO and the CNC shield with DRV8825 drivers. I have the Microstepping set ot 1/16th. May be the microstepping needs adjusting on your drivers.

**Comment by StuartB4** [comment]
One other thing. Is this an anti-backlash fitting or just a threaded fitting. 

![image](https://user-images.githubusercontent.com/16980026/110656564-f7b89700-81b7-11eb-8637-dd6ea2b210fe.png)

**Reply by khturnings** [reply]
That's a standard leadscrew.
You need something like this that fits the thread you have.
https://images-na.ssl-images-amazon.com/images/I/71mvV7RdYWL._AC_SL1500_.jpg

**Reply by dwaterford75** [reply]
What difference will an anti-backlash nut make?  A bit more accurate but surely it won't fix my problem.
